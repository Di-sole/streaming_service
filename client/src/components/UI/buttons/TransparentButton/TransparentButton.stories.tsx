import type { Meta, StoryObj } from '@storybook/react';
import TransparentButton from './TransparentButton';
import './TransparentButton.module.scss';

const meta: Meta<typeof TransparentButton> = {
    title: 'Button/TransparentButton',
    component: TransparentButton,
    tags: ['autodocs'],
    argTypes: {
        textColor: {
            type: 'string',
            descriptrion: 'цвет шрифта',
            defaultValue: 'bright',
            options: ['bright', 'faded'],
        },
    },
    decorators: [
        (Story) => (
            <div
                style={{
                    background: '#100e19',
                    height: '50px',
                    display: 'flex',
                    alignItems: 'center',
                    paddingLeft: '10px',
                }}
            >
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof TransparentButton>;

export const BrightButton: Story = {
    args: {
        children: 'bright button',
        textColor: 'bright',
    },
};

export const FadedButton: Story = {
    args: {
        children: 'faded button',
        textColor: 'faded',
    },
};
