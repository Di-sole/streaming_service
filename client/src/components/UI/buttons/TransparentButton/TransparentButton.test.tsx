import { render, screen } from '@testing-library/react';
import TransparentButton from './TransparentButton';

describe('TRANSPARENTBUTTON SNAPSHOTS TESTS', () => {
    test('textColor="bright" TransparentButton should not be changed', () => {
        render(<TransparentButton className="button" textColor="bright" children="Text" />);
        const button = screen.getByRole('button');

        expect(button).toMatchSnapshot();
    });

    test('textColor="faded" TransparentButton should not be changed', () => {
        render(<TransparentButton className="button" textColor="faded" children="Text" />);
        const button = screen.getByRole('button');

        expect(button).toMatchSnapshot();
    });
});
