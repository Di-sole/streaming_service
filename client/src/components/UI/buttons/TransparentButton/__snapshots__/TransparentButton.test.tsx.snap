// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TRANSPARENTBUTTON SNAPSHOTS TESTS textColor="bright" TransparentButton should not be changed 1`] = `
<button
  class="button button_bright button"
  type="button"
>
  Text
</button>
`;

exports[`TRANSPARENTBUTTON SNAPSHOTS TESTS textColor="faded" TransparentButton should not be changed 1`] = `
<button
  class="button button_faded button"
  type="button"
>
  Text
</button>
`;
